src/pynwb/__init__.py

```diff
@@ -261,7 +261,8 @@ def __init__(self, **kwargs):
             popargs('path', 'mode', 'manager', 'extensions', 'load_namespaces',
                     'file', 'comm', 'driver', 'herd_path', kwargs)
         # Define the BuildManager to use
-        if mode in 'wx' or manager is not None or extensions is not None:
+        io_modes_that_create_file = ['w', 'w-', 'x']
+        if mode in io_modes_that_create_file or manager is not None or extensions is not None:
```

Comment:
My style is to be a bit more compact:
```suggestion
        if mode in ['w', 'w-', 'x'] or manager is not None or extensions is not None:
```

but this is fine if you think it's more clear

Reply:
I prefer to err on the side of being too explicit. I feel that the consiness of the previous code is what allowed the bug to slip by:

```python
            if 'w' in mode or mode == 'x':
```
So if you are fine with it I would like to keep it as it is. The variable name works as documentation of intent.